libcontainerd/client_linux.go

```diff
@@ -563,6 +563,8 @@ func (clnt *client) Restore(containerID string, options ...CreateOption) error {
 		clnt.remote.Lock()
 		return nil
 	}
+	// relock because of the defer
+	clnt.remote.Lock()
```

Comment:
That's asking the reader to understand in which order `defer` statements are executed :stuck_out_tongue: Seems more complicated to me.